Sources/ParseSwift/API/Responses.swift

```diff
@@ -33,3 +33,15 @@ internal struct UpdateResponse: Decodable {
         return object
     }
 }
+
+internal struct FetchResponse: Decodable {
```

Comment:
My last commit may look like a big change, but it's mostly in the test cases. Basically I started using the getTestDecoder() I added in one of the previous commits so I can get dates in the correct format (from the dateEncodingStrategy). This allows us to compare the mocked dates to the Parse-Swift ones. Should also verify more of the ParseEncoder.

The more important part of the commit is the addition of `FetchResponse` to `SaveResponse` and `UpdateResponse`. Originally I was thinking why not just apply createdAt, updatedAt, and objectId to all objects pulled from the parse-server, but I believe I know why these are there. The business logic (createdAt, updatedAt, and objectId) is meant to all come from the parse-server and should be immutable (or almost immutable, things get strange if you updated createdAt or objectId via Parse Dashboard or some other means). The three types of responses ensure the only the correct business logic is updated locally by the API. I would imagine this is important for the local storage stuff you are working on as once the createdAt and objectId are set, they should essentially never change.

Reply:
That makes sense! I think it's a good idea to defer to the server for this stuff. That being said, there's one or two things I'd like to point out:
* `SaveOrUpdateResponse` doesn't include `FetchResponse`, and `SaveOrUpdateOrFetchResponse` is a little cumbersome. So maybe `SaveOrUpdateResponse` should be renamed to `WriteResponse`?
* I don't like the signature for `apply`. I think it should be redefined so the signature reads `applied(to:)`. Since we're not mutating the object directly, it should be `applied` instead of `apply`, and the `to` just makes it read nicer. Alternatively, we could make it take the object as an inout parameter, and change it to `apply(to:)`.

Reply:
Oic, you were referring to the 'SaveOrUpdateResponse` name not including "fetch", I'm fine with `WriteResponse`. 

I made your suggested changes to "apply(to:)"